### ISSUE TITLE
Add ONBUILDEND support to Dockerfile

### ISSUE THREAD

**Issue by Yajo** [moby/moby#29910]
**Description**

There's support for `ONBUILD` commands, that trigger before starting the execution of the current (child) Dockerfile.

It would be great to **add `ONBUILDEND`, that trigger after building** the child Dockerfile.

This would be useful for cases where you expect somebody to `COPY` some folder somewhere before actually i.e. compiling the code:

Parent Dockerfile:

```
FROM alpine
RUN apk add --no-cache python
ONBUILDEND pip install --no-cache-dir --editable /usr/src/app
```

Child Dockerfile:

```
FROM previousimage
COPY mypackage /usr/src/app
```

This way you let the subproject have the directory structure they desire, [instead of e.g. forcing them to put the `requirements.txt` file in the project root folder](https://github.com/docker-library/python/blob/7663560df7547e69d13b1b548675502f4e0917d1/2.7/onbuild/Dockerfile#L6).

**Comment by thaJeztah** [comment]
With the classic builder being largely frozen in favour of buildkit, and the Dockerfile spec now living in that repository, I'm closing this issue here, but feel free to open an new issue in the buildkit repository for discussion (https://github.com/moby/buildkit). the `ONBUILD` option has historically been quite tricky to get "right", so I'm not sure how much interest there is to implement this (but could be proposed as experimental option)
